ids/node_id.go

```diff
@@ -4,54 +4,46 @@
 package ids
 
 import (
-	"bytes"
-	"errors"
 	"fmt"
 
 	"github.com/ava-labs/avalanchego/staking"
 	"github.com/ava-labs/avalanchego/utils"
+	"github.com/ava-labs/avalanchego/utils/cb58"
 	"github.com/ava-labs/avalanchego/utils/hashing"
 )
 
-const (
-	NodeIDPrefix = "NodeID-"
-	NodeIDLen    = ShortIDLen
-)
-
 var (
-	EmptyNodeID = NodeID{}
+	EmptyNodeID = NodeID(string(EmptyShortNodeID[:]))
 
-	errShortNodeID = errors.New("insufficient NodeID length")
-
-	_ utils.Sortable[NodeID] = NodeID{}
+	_ utils.Sortable[NodeID] = (*NodeID)(nil)
 )
 
-type NodeID ShortID
-
-func (id NodeID) String() string {
-	return ShortID(id).PrefixedString(NodeIDPrefix)
-}
+type NodeID string
 
 // Any modification to Bytes will be lost since id is passed-by-value
 // Directly access NodeID[:] if you need to modify the NodeID
-func (id NodeID) Bytes() []byte {
-	return id[:]
+func (nodeID NodeID) Bytes() []byte {
+	return []byte(nodeID)
 }
 
-func (id NodeID) MarshalJSON() ([]byte, error) {
-	return []byte("\"" + id.String() + "\""), nil
+func (nodeID NodeID) String() string {
```

Comment:
We can get rid of `String` because this is already a string

Reply:
Not exactly.
`NodeID` is a string. I'd like it to be a byte slice, but I can't use byte slices as map key (not comparable), so I use a string.
`NodeID` holds node bytes into a string; String() method encode the bytes content and prepend a prefix.